src/full/Agda/Interaction/Options/Lenses.hs

```diff
@@ -131,6 +133,14 @@ modifySafeMode = modifyTC . mapSafeMode
 putSafeMode :: SafeMode -> TCM ()
 putSafeMode = modifyTC . setSafeMode
 
+isBuiltinWithSafePostulates :: FilePath -> TCM Bool
+isBuiltinWithSafePostulates file = do
+  libdir <- liftIO defaultLibDir
+  let unsafeBuiltins =
+        [ libdir </> "prim" </> "Agda" </> "Builtin" </> "TrustMe.agda"
+        , libdir </> "prim" </> "Agda" </> "Builtin" </> "Equality" </> "Rewrite.agda"
+        ]
+  return $ (prefixPath libdir file) && not (elem file unsafeBuiltins)
```

Comment:
It would be a bit more tedious but also a bit more bug-proof to list the
files for which the postulates are considered safe rather than assuming
all `prim` files are safe unless explicitly excluded.

Reply:
Yes, probably a good idea. Maybe it would also be good to add a test that all builtin files are either safe or unsafe wrt postulates.